src/transformers/models/wav2vec2/modeling_wav2vec2.py

```diff
@@ -1034,6 +1052,38 @@ def forward(self, hidden_states):
         return hidden_states
 
 
+class Wav2Vec2AttnAdapterLayer(nn.Module):
+    def __init__(self, config):
+        """
+        Implements adapter modules directly with 3D tensor weight as parameters and without using ModuleList orto speed
+        up training throughput.
+        """
+        super().__init__()
+        self.adapter_num = config.num_attn_adapters
+        self.input_dim = 16
```

Comment:
Are we good hard-coding this? Or better to set it based on the dimensionality of the `hidden_states` in case the model dims are changed?

Reply:
Yep agree! Was also thinking a config param would be good here (one that's tied to the `hidden_states` dim)

Reply:
Moved it to the config